web-marketplace/src/components/organisms/ChooseCreditsForm/ChooseCreditsForm.tsx

```diff
@@ -144,11 +161,14 @@ export function ChooseCreditsForm({
   const handlePaymentOptions = useCallback(
     (option: string) => {
       setPaymentOption(option as PaymentOptionsType);
-      form.setValue(CREDIT_VINTAGE_OPTIONS, []);
-      form.setValue(
-        CURRENCY,
-        option === PAYMENT_OPTIONS.CARD ? cardCurrency : defaultCryptoCurrency,
-      );
+      form.reset({
```

Comment:
why do we need to do this?

Reply:
If we don't reset the form, then the form validation, now depending on the `paymentOption` param, doesn't get updated with the newly selected option handled by `handlePaymentOptions`.

Reply:
if you just want to trigger validation you could use `shouldValidate` or `trigger`

Reply:
Is not that we need to trigger validation, we want the validation scheme to get updated with the new `paymentOption` value. If I don't use `reset` then I see the `userBalance` error in the USD form. Tried with `shouldValidate` and `trigger` but didn't work